fe/fe-core/src/main/java/org/apache/doris/nereids/parser/Dialect.java

```diff
@@ -0,0 +1,117 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.parser;
+
+import javax.annotation.Nullable;
+
+/**
+ * ParseDialect enum, maybe support other dialect.
+ */
+public enum Dialect {
+    /**
+     * Doris parser dialect
+     */
+    DORIS("doris"),
+    /**
+     * Trino parser dialect
+     */
+    TRINO("trino"),
+    /**
+     * Presto parser dialect
+     */
+    PRESTO("presto"),
+    /**
+     * Spark sql parser dialect
+     */
+    SPARK_SQL("spark_sql"),
+    /**
+     * Hive parser dialect
+     */
+    HIVE("hive"),
+    /**
+     * Iceberg parser dialect
+     */
+    ICEBERG("iceberg"),
```

Comment:
iceberg\paimon\hudi\dlf are not query engine, so they don't have sql dialect, remove them